charts/datadog/ci/otel-agent-gateway-hpa-values.yaml

```diff
@@ -0,0 +1,51 @@
+targetSystem: "linux"
+fullnameOverride: "gw-only"
+agents:
+  enabled: false
+clusterAgent:
+  enabled: false
+otelAgentGateway:
+  enabled: true
+  ports:
+    - containerPort: "4317"
+      name: "otel-grpc"
+  config: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: "0.0.0.0:4317"
+    exporters:
+      debug:
+      otlphttp:
+        endpoint: otlp.datadoghq.com
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp]
+          exporters: [otlphttp, debug]
+        metrics:
+          receivers: [otlp]
+          exporters: [otlphttp, debug]
+        logs:
+          receivers: [otlp]
+          exporters: [otlphttp, debug]
+  replicas: 4
+  autoscaling:
+    enabled: true
+    annotations: {}
+    minReplicas: 2
+    maxReplicas: 10
+    metrics:
+      # Aim for high CPU utilization for higher throughput
+      - type: Resource
+        resource:
+          name: cpu
+          target:
+            type: Utilization
+            averageUtilization: 80
```

Comment:
I think we might want to look at our SMP tests to check memory evolution as a function of CPU under basic use-cases.

I believe throughout the PR we establish CPU-bound workloads in GW (which are the likely reality), but I'm wondering about the possibility of memory bound workloads too, in which case scaling on CPU and not memory might not be ideal.